zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -219,6 +219,14 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
     // Check if user belongs to super user group
     if (clientId.equals(superUser)) {
       newAuthIds.add(new Id(X509AuthenticationUtil.SUPERUSER_AUTH_SCHEME, clientId));
+    } else if (superUserDomainNames.stream().anyMatch(d -> domains.contains(d))) {
+      domains.forEach(d -> {
```

Comment:
Could you please use the stream filter method to simplify this part?
`List<T> intersect = list1.stream()
                         .filter(list2::contains)
                         .collect(Collectors.toList());`
Maybe we can also use the condition `!intersect.isEmpty()` as the if condition in this clause